pytket/binders/circuit/main.cpp

```diff
@@ -74,6 +93,18 @@ PYBIND11_MODULE(circuit, m) {
           py::arg("latex") = false)
       .def("__eq__", &Op::operator==)
       .def("__repr__", [](const Op &op) { return op.get_name(); })
+      .def(
+          "free_symbols",
+          [](const Op *op) {
+            SymSet symbols;
+            const auto &gate = static_cast<const Gate &>(*op);
```

Comment:
I don't see why we need to cast to a `Gate` (and the `static_cast` is dangerous anyway). The `free_symbols()` method is defined for any `Op`; it will just throw an exception in some cases.

Reply:
Actually that was my initial solution. But I felt it made more sense with the custom exception. On hindsight, this is less generic and would throw another error than the one we are interested in. Will revert.